src/app/modules/InstanceDrawer/InstanceDrawer.tsx

```diff
@@ -49,6 +75,49 @@ const InstanceDrawer: VoidFunctionComponent<InstanceDrawerProps> = ({
     noInstances,
   } = useInstanceDrawer();
 
+  const {
+    id: kafkaId,
+    cloud_provider: provider,
+    region,
+    size_id,
+    instance_type,
+  } = instanceDrawerInstance || {};
+
+  const fetchAvailableSizes = useCallback(async () => {
+    const api = new DefaultApi(
+      new Configuration({
+        accessToken: getToken(),
+        basePath,
+      })
+    );
+
+    if (provider && region) {
+      try {
+        setIsLoadingSize(true);
+
+        const sizes = await api.getInstanceTypesByCloudProviderAndRegion(
```

Comment:
Where this comes from? Do you mind using the same hook from #1148  PR for consistency

Reply:
I was thinking about it but avoid due to having quota check in useGetAvailableSizes hook. 
We need to lift quota code so that reuse useGetAvailableSizes hook.

Thanks for suggestions.

Reply:
That is no problem actually. We can have nested hooks so:

- UseQuota
- UseSizes
- UseSizesWithQuota

This is not limiting factor for anything, just small refactor is needed. (Although Riccardo might want to refactor this entire thing anyway to state machine later on)